app/gui/view/src/project.rs

```diff
@@ -26,6 +27,45 @@ use ensogl_hardcoded_theme::Theme;
 
 
 
+// ============================
+// === WayOfOpeningSearcher ===
+// ============================
+
+/// An enum describing how the searcher was opened.
+#[derive(Clone, CloneRef, Copy, Debug, PartialEq)]
+pub enum WayOfOpeningSearcher {
```

Comment:
I'd rename it to `ComponentBrowserOpenReason`. The "way" word is broader than "reason" IMO

Reply:
Btw, we should lowly rename "searcher" to component browser in the future. I think that new apis can already use the new name.

Reply:
I think `Cause` is even better.